src/Microsoft.OData.Core/UriParser/Binders/SelectExpandBinder.cs

```diff
@@ -441,6 +441,8 @@ private SelectItem GenerateExpandItem(ExpandTermToken tokenIn)
 
             ODataExpandPath pathToNavProp = new ODataExpandPath(pathSoFar);
 
+            IEdmTypeReference elementTypeReference = hasDerivedTypeSegment ? derivedType.ToTypeReference() : null;
+
             // $apply
             ApplyClause applyOption = BindApply(tokenIn.ApplyOptions, this.ResourcePathNavigationSource, targetNavigationSource);

```

Comment:
>BindApply [](http://example.com/codeflow?start=38&length=9)

should we be passing elementTypeReference to BindApply?

Reply:
Yes - I have fixed the code.